src/authorization-control.ts

```diff
@@ -274,4 +275,47 @@ export class AuthorizationControlApi {
     }
     return { success: !resp?.payload?.errorCode, body: { ...resp.payload } }
   }
+
+  /*
+   * Function to take an Authorization Control Merchant Identifier Exemption
+   * token, and an active status argument, send that to Marqeta, and have an
+   * Authorization Control Merchant Identifier Exemption's active status updated.
+   * If not active status argument is supplied, the default active status is set
+   * to false.
+   */
+  async updateMerchantStatus(merchant: Partial<Merchant>): Promise<{
+    success: boolean,
+    body?: Merchant,
+    error?: MarqetaError,
+  }> {
+    const {
+      token,
+      active = false,
+    } = merchant
+    if (!token) {
+      return {
+        success: true,
+        body: {
+          name: merchant.name ?? '',
+        }
```

Comment:
Why is the return error component called `body`? As if it's the `body` of a `POST` or something... I'm not at all sure what you're going for... can you help me understand?

Also, why, if this is an error, aren't you returning the `merchant`, and why just the `name` - and even that, as an empty string if it's not there? Why not just return:

```typescript
       return {
         success: true,
         merchant,
       }
```
??

-----

But then is this not an error? And if it's an error - why are you returning `success: true` as opposed to:

```typescript
       return {
         success: false,
         error: 'This call requires a Merchant to have an active "token" - and this one does not.',
         merchant,
       }
```

then it's clear what the error was, that you found it, etc. I'd probably put it into the proper `MarqetaError` format with `mkError()` or something...

Reply:
good point